rclcpp_lifecycle/src/state.cpp

```diff
@@ -14,47 +14,90 @@
 
 #include "rclcpp_lifecycle/state.hpp"
 
-#include <lifecycle_msgs/msg/state.hpp>
+#include <string>
 
-#include <rcutils/allocator.h>
-#include <rcutils/strdup.h>
-#include <rcl_lifecycle/data_types.h>
+#include "lifecycle_msgs/msg/state.hpp"
 
-#include <string>
+#include "rcutils/allocator.h"
+#include "rcutils/strdup.h"
+
+#include "rcl_lifecycle/data_types.h"
 
 namespace rclcpp_lifecycle
 {
 
-State::State()
-: State(lifecycle_msgs::msg::State::PRIMARY_STATE_UNKNOWN, "unknown")
+State::State(rcutils_allocator_t allocator)
+: State(lifecycle_msgs::msg::State::PRIMARY_STATE_UNKNOWN, "unknown", allocator)
 {}
 
-State::State(uint8_t id, const std::string & label)
-: owns_rcl_state_handle_(true)
+State::State(
+  uint8_t id,
+  const std::string & label,
+  rcutils_allocator_t allocator)
+: allocator_(allocator),
+  owns_rcl_state_handle_(true),
+  state_handle_(nullptr)
 {
   if (label.empty()) {
     throw std::runtime_error("Lifecycle State cannot have an empty label.");
   }
 
-  auto state_handle = new rcl_lifecycle_state_t;
+  auto state_handle = reinterpret_cast<rcl_lifecycle_state_t *>(
+    allocator_.allocate(sizeof(rcl_lifecycle_state_t), allocator_.state));
```

Comment:
The return value of `allocate` needs to be checked for `NULL`.

Same below.

Reply:
addressed in commit https://github.com/ros2/rclcpp/pull/395/commits/dba6601f33c30834d1c9a05f1d577696723ccfad